src/lib.rs

```diff
@@ -0,0 +1,704 @@
+		let storage_dir_path = "/tmp/ldk_lite/".to_string();
+		let esplora_server_url = "https://blockstream.info/api".to_string();
+		let network = bitcoin::Network::Testnet;
+		let listening_port = 9735;
+
+		let config =
+			LdkLiteConfig { storage_dir_path, esplora_server_url, network, listening_port };
+
+		Self { config }
+	}
+
+	/// Creates a new builder instance from an [`LdkLiteConfig`].
+	pub fn from_config(config: LdkLiteConfig) -> Self {
+		Self { config }
+	}
+
+	/// Sets the used storage directory path.
+	///
+	/// Default: `/tmp/ldk_lite/`
+	pub fn storage_dir_path(&mut self, storage_dir_path: String) -> &mut Self {
+		self.config.storage_dir_path = storage_dir_path;
+		self
+	}
+
+	/// Sets the Esplora server URL.
+	///
+	/// Default: `https://blockstream.info/api`
+	pub fn esplora_server_url(&mut self, esplora_server_url: String) -> &mut Self {
+		self.config.esplora_server_url = esplora_server_url;
+		self
+	}
+
+	/// Sets the Bitcoin network used.
+	///
+	/// Options: `mainnet`/`bitcoin`, `testnet`, `regtest`, `signet`
+	///
+	/// Default: `testnet`
+	pub fn network(&mut self, network: String) -> &mut Self {
+		self.config.network = match network.as_str() {
+			"mainnet" => bitcoin::Network::Bitcoin,
+			"bitcoin" => bitcoin::Network::Bitcoin,
+			"testnet" => bitcoin::Network::Testnet,
+			"regtest" => bitcoin::Network::Regtest,
+			"signet" => bitcoin::Network::Signet,
+			_ => bitcoin::Network::Testnet,
+		};
+		self
+	}
+
+	/// Sets the port on which [`LdkLite`] will listen for incoming network connections.
+	///
+	/// Default: `9735`
+	pub fn listening_port(&mut self, listening_port: u16) -> &mut Self {
+		self.config.listening_port = listening_port;
+		self
+	}
+
+	/// Builds an [`LdkLite`] instance according to the options previously configured.
+	pub fn build(self) -> Result<LdkLite, Error> {
+		let config = Arc::new(self.config);
+
+		let ldk_data_dir = format!("{}/ldk", &config.storage_dir_path.clone());
+		fs::create_dir_all(ldk_data_dir.clone())?;
+
+		let bdk_data_dir = format!("{}/bdk", config.storage_dir_path.clone());
+		fs::create_dir_all(bdk_data_dir.clone())?;
+
+		// Step 0: Initialize the Logger
+		let logger = Arc::new(FilesystemLogger::new(config.storage_dir_path.clone()));
+
+		// Step 1: Initialize the on-chain wallet and chain access
+		let seed = util::read_or_generate_seed_file(Arc::clone(&config))?;
+		let xprv = bitcoin::util::bip32::ExtendedPrivKey::new_master(config.network, &seed)?;
+
+		let bdk_wallet = bdk::Wallet::new(
+			bdk::template::Bip84(xprv.clone(), bdk::KeychainKind::External),
+			Some(bdk::template::Bip84(xprv.clone(), bdk::KeychainKind::Internal)),
+			config.network,
+			MemoryDatabase::default(),
+		)?;
+
+		// TODO: Check that we can be sure that the Esplora client re-connects in case of failure
+		// and and exits cleanly on drop. Otherwise we need to handle this/move it to the runtime?
+		let blockchain = EsploraBlockchain::new(&config.esplora_server_url, 20).with_concurrency(8);
```

Comment:
It depends on the configured implementation of the `EsploraClient`, i.e., the underlying HTTP lib. I currently use the blocking variant based on `ureq`, which indeed spawns a new thread for each request. Here I simply doubled the default concurrency value of 4 to speed up the wallet sync a bit, but not sure what would be optimal for the user. We probably don't want to introduce a background process that spikes the load and bandwidth usage in regular intervals..

Reply:
Yes, this should be possible, since the Esplora client even directly supports this via `reqwest`. I even implemented the necessary API calls on that side too. However, the BDK branch I'm currently working on is only half done with ripping out the Esplora client from the BDK main project. As BDK does some magic via features+macros to replace the blocking/async variants, which I don't want to reproduce in my code, I opted for the blocking variant for the time being. So, another topic I intend to re-visit soonish, when BDKs restructuring has progressed further.

Reply:
Nope, `reqwest` or `ureq` it is for the time beeing.